### ISSUE TITLE
KRM for private marketplace / service catalog 

### ISSUE THREAD

**Issue by fmichaelobrien** [GoogleCloudPlatform/pubsec-declarative-toolkit#291]
Priority for 2023 - shadow tracking in https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/issues/250
references
I will create a PM and/or a service catalog in one of my orgs to verify
https://cloud.google.com/marketplace/docs/governance/enable-private-marketplace
https://cloud.google.com/service-catalog
https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/blob/main/docs/google-cloud-security-controls.md#marketplace-role-restricted

Org policies to validate
- Restrict access on marketplace services constraints/commerceorggovernance.marketplaceServices
- on top of already set constraints/commerceorggovernance.disablePublicMarketplace

**Comment by obriensystems** [comment]
see similar #293

**Comment by obriensystems** [comment]
By default the org policy at the organization level when set to enforced - has no effect yet
constraints/commerceorggovernance.marketplaceServices

we can still create services normally either via console or gcloud cli.
The workaround is to remove billing access to particular users for now - see #290 

<img width="932" alt="Screenshot 2023-02-27 at 21 00 54" src="https://user-images.githubusercontent.com/24765473/221733613-c3d80f0f-d269-4541-95e5-34d948d49631.png">
for example creating a GKE cluster
<img width="1229" alt="Screenshot 2023-02-27 at 21 02 58" src="https://user-images.githubusercontent.com/24765473/221733899-a7e8209f-4dcd-468d-bbc5-fda0870de49a.png">

**Comment by KingBain** [comment]
I'm sure this is confusion on my part but the private marketplace and the  service catalog are two seperate an unique offerings right. 

My understanding is the private market place is a curated list of pre-approved public marketplace offerings and the service catalog is a catalog of turn key deploy-able templates(sometimes customized)

**Comment by KingBain** [comment]
@obriensystems you might have to confirm on a new organization that hasnt already enabled API's and services. We're experiencing an enabled private market place(thats empty) thats blocking the enabling of services like cloud run and compute via the console. I havent tried your workaround of enabling via cli yet.  

I'm curious in your findings about the workaround method.

**Comment by obriensystems** [comment]
You are right, I would expect that disabling the public marketplace would disable GCP service creation - as services come in through the marketplace under the covers.  I have had the flag on for a while and ran through 50+ project creations with additional service enablements - see for example
https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/blob/canary/solutions/document-processing/gcloud/deployment.sh

Check on default services like bigquery that get enabled by default on new projects - where compute does not
I created a new org a couple days ago but it already has 5 projects - i will create another one and set the mp flag first
Also I hope there is not different behavior for org admins/owners that override the policy - should not be

I'll try to get the to the bottom of this - as I have not seen any effect of the flag since Nov and was planning on following up

https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/issues/205
https://github.com/canada-ca/accelerators_accelerateurs-gcp/issues/52

The link to #290 is only for a secondary workaround for GR 12 that effectively disables paid 3rd party marketplace services

**Comment by cartyc** [comment]
There is a second marketplace policy which might help with this issue.
Policy:
```
constraints/commerceorggovernance.marketplaceServices
```
Description:
```
This list constraint defines the set of services allowed for marketplace organizations, and can only include values from the list below:
PRIVATE_MARKETPLACE
IAAS_PROCUREMENT
If PRIVATE_MARKETPLACE is in the allowed value list, the private marketplace is enabled. If the IAAS_PROCUREMENT is in the allowed value list, the IaaS procurement governance experience is enabled for all products. By default, the private marketplace is disabled and the IaaS procurement governance experience is disabled.
```

**Comment by fmichaelobrien** [comment]
sounds good, did a check before for possible policies and so far we have these 2 policies - I have not tested out the 2nd marketplaceServices yet.
https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/issues/291#issue-1601815378

Org policies to validate

Restrict access on marketplace services constraints/commerceorggovernance.marketplaceServices
on top of already set constraints/commerceorggovernance.disablePublicMarketplace

Will also need to track down under what conditions disablePublicMarketplace takes effect (so far previously used orgs do not kick in the restriction when set) - trying another completely clean org....

**Comment by fmichaelobrien** [comment]
Revisit the lockdown on the org policy and validate (we last used it right after release)
[2:04 PM] Michael O'Brien - Google (Guest)
the org policy had issues (no effect in the past) - need to retest
Set to true
constraints/commerceorggovernance.disablePublicMarketplace
https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/blob/main/docs/google-cloud-security-controls.md#12-configuration-of-cloud-marketplaces
We used to use billing permissions as a backup
Starting with the doc up
 
[2:04 PM] Michael O'Brien - Google (Guest)
Set to true
constraints/commerceorggovernance.disablePublicMarketplace
[2:05 PM] Michael O'Brien - Google (Guest)
https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/issues/170
[2:06 PM] Michael O'Brien - Google (Guest)
private catalog I would think - checking jiras
[2:07 PM] Michael O'Brien - Google (Guest)
https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/issues/250
shadow for kcc
https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/issues/291
[2:08 PM] Michael O'Brien - Google (Guest)
constraints/commerceorggovernance.marketplaceServices



best practices for managing a private marketplace - verify the cloud broker check

**Comment by obriensystems** [comment]
retest for scrum 20240207
